src/m365/spo/commands/commandset/commandset-get.ts

```diff
@@ -0,0 +1,156 @@
+import { formatting } from '../../../../utils/formatting';
+import { spo } from '../../../../utils/spo';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  title?: string
+  id?: string;
+  clientSideComponentId?: string;
+  scope?: string;
+}
+
+class SpoCommandSetGetCommand extends SpoCommand {
+  private static readonly scopes: string[] = ['All', 'Site', 'Web'];
+  private static readonly baseLocation: string = 'ClientSideExtension.ListViewCommandSet';
+  private static readonly allowedCommandSetLocations: string[] = [SpoCommandSetGetCommand.baseLocation, `${SpoCommandSetGetCommand.baseLocation}.CommandBar`, `${SpoCommandSetGetCommand.baseLocation}.ContextMenu`];
+
+  public get name(): string {
+    return commands.COMMANDSET_GET;
+  }
+
+  public get description(): string {
+    return 'Get a ListView Command Set that is added to a site.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        title: typeof args.options.title !== 'undefined',
+        id: typeof args.options.id !== 'undefined',
+        clientSideComponentId: typeof args.options.clientSideComponentId !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-t, --title [title]'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-c, --clientSideComponentId [clientSideComponentId]'
+      },
+      {
+        option: '-s, --scope [scope]',
+        autocomplete: SpoCommandSetGetCommand.scopes
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (typeof isValidUrl === 'string') {
+          return isValidUrl;
+        }
+
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID.`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID.`;
+        }
+
+        if (args.options.scope && SpoCommandSetGetCommand.scopes.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Valid scopes are ${SpoCommandSetGetCommand.scopes.join(', ')}`;
+        }
+
+        return true;
```

Comment:
I suppose we may just move the URL validation to be done as last and have a bit less code 🤔

```suggestion
        if (args.options.id && !validation.isValidGuid(args.options.id)) {
          return `${args.options.id} is not a valid GUID.`;
        }

        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
          return `${args.options.clientSideComponentId} is not a valid GUID.`;
        }

        if (args.options.scope && SpoCommandSetGetCommand.scopes.indexOf(args.options.scope) < 0) {
          return `${args.options.scope} is not a valid scope. Valid scopes are ${SpoCommandSetGetCommand.scopes.join(', ')}`;
        }

        return validation.isValidSharePointUrl(args.options.webUrl);
```